File: Stronghold.Core/Entities/Supplier.cs
namespace Stronghold.Core.Entities;

public class Supplier : BaseEntity
{
    public string Name { get; set; } = null!;
    public string ContactEmail { get; set; } = null!;
    public string ContactPhone { get; set; } = null!;

    public ICollection<Supplement> Supplements { get; set; } = new List<Supplement>();
}

File: UI/stronghold_desktop/lib/models/supplier.dart
class Supplier {
  final int id;
  final String name;
  final String contactEmail;
  final String contactPhone;

  Supplier({
    required this.id,
    required this.name,
    required this.contactEmail,
    required this.contactPhone,
  });

  factory Supplier.fromJson(Map<String, dynamic> json) => Supplier(
        id: json['id'] as int,
        name: json['name'] as String,
        contactEmail: json['contactEmail'] as String,
        contactPhone: json['contactPhone'] as String,
      );
}
